lib/data_integrity_utils.pm

```diff
@@ -32,9 +32,21 @@ sub verify_checksum {
         my $checksum = get_required_var $image;
         $image =~ s/CHECKSUM_//;
         my $image_path = get_required_var $image;
+        my $dir_path;
```

Comment:
$dir_path is re-declared here. If you don't want to pass the path from the caller I would suggest to create function get_path and delete the parameter.